File: nuido_flow_ai/flows/ai/time_ai_mcp_node.py
# THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
#
# This software is released under the MIT License.
# https://opensource.org/licenses/MIT
#
# THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
# DO NOT USE IT IN PRODUCTION.
from odoo.addons.nuido_flow.flows.core.base_node import BaseNode
from autogen_ext.tools.mcp import StdioServerParams

class TimeAiMcpNode(BaseNode):
    def _process(self, params):
        tz = self.env.context["user"].tz
        mcp_server = StdioServerParams(command="uvx", args=["mcp-server-time", f"--local-timezone={tz}"])
        return {
            "mcp_server": mcp_server
        }

File: nuido_flow_ai/static/src/models/ai/time_ai_mcp_node.js
// THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT
//
// THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
// DO NOT USE IT IN PRODUCTION.
import { NodeModel } from "@nuido/models/node";
import { AiMcpPort } from "@nuido_flow_ai/components/ports/ai_mcp_port";
export class TimeAiMcpNodeModel extends NodeModel {
    setup() {
        const auxOutId = "aux-out-" + this.id + "-time-ai-mcp";
        this.addAuxOutPort(auxOutId, AiMcpPort.name, Number.MAX_SAFE_INTEGER, {
            role: "ai-mcp"
        });
    }
}
